docs/tracingpolicy/README.md

```diff
@@ -0,0 +1,1193 @@
+`[`binary == /bin/cat `AND` arg1 == /etc/shadow `AND` NetNs is host`]`
+
+We can modify the previous example as follows:
+
+> Generate a kprobe event if `/etc/shadow` was opened by `/bin/cat` which has
+> host `Net` and `Mnt` namespace access
+
+```yaml
+apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "example_ns_2"
+spec:
+  kprobes:
+    - call: "fd_install"
+      syscall: false
+      args:
+        - index: 0
+          type: int
+        - index: 1
+          type: "file"
+      selectors:
+        - matchBinaries:
+          - operator: "In"
+            values:
+            - "/bin/cat"
+          matchArgs:
+          - index: 1
+            operator: "Equal"
+            values:
+            - "/etc/shadow"
+          matchNamespaces:
+          - namespace: Mnt
+            operator: In
+            values:
+            - "host_ns"
+          - namespace: Net
+            operator: In
+            values:
+            - "host_ns"
+```
+
+Here we have a single selector. This CRD will match if:
+
+`[`binary == /bin/cat `AND` arg1 == /etc/shadow `AND` `(`MntNs == host `AND`
+NetNs == host`)` `]`
+
+#### Capabilities filter
+
+Capabilities filters can be specified under the `matchCapabilities` field and
+provide filtering of calls based on Linux capabilities in the specific sets.
+
+An example syntax is:
+```yaml
+- matchCapabilities:
+  - type: Effective
+    operator: In
+    values:
+    - "CAP_CHOWN"
+    - "CAP_NET_RAW"
+```
+
+This will match if: [`Effective` capabilities contain `CAP_CHOWN`] OR
+[`Effective` capabilities contain `CAP_NET_RAW`]
+
+- `type` can be: `Effective`, `Inheritable`, or `Permitted`.
+- `operator` can be `In` or `NotIn`
+- `values` can be any supported capability. A list of all supported
+  capabilities can be found in `/usr/include/linux/capability.h`.
+
+**Limitations**
+
+1. There is no limit in the number of capabilities listed under `values`.
+2. Only one `type` field can be specified under `matchCapabilities`.
+
+#### Namespace changes filter
+
+Namespace changes filter can be specified under the `matchNamespaceChanges`
+field and provide filtering based on calls that are changing Linux namespaces.
+This filter can be especially useful to track container escape or execution of
+code in a new namespace, for example using user namespace to exploit root-only
+kernel features.
+
+```yaml
+matchNamespaceChanges:
+- operator: In
+  values:
+  - "Mnt"
+```
+
+The `unshare` command, or executing in the host namespace using `nsenter` can
+be used to test this feature. See a
+[demonstration example](https://github.com/cilium/tetragon/blob/c511bf632ee58d401aa0f0cf1df50b33760515e3/crds/examples/match_namespace_changes.yaml)
+of this feature.
+
+#### Capability changes filter
+
+Capability changes filter can be specified under the `matchCapabilityChanges`
+field and provide filtering based on calls that are changing Linux capabilities.
```

Comment:
The previous comment applies here as well. What's the logic for keeping track of cap changes? we know what capabilities the process has and we match against a "delta" when a certain system call is invoked?

Reply:
Similarly:
https://github.com/cilium/tetragon/blob/1a68bf2681e79ec57c37863b1e5b809e7be04413/bpf/process/pfilter.h#L284-L297

Reply:
I added the sentences about the logic!